sklearn/neural_network/multilayer_perceptron.py

```diff
@@ -939,6 +939,30 @@ def predict(self, X):
 
         return self._label_binarizer.inverse_transform(y_pred)
 
+    def fit(self, X, y):
+        """Fit the model to data matrix X and target(s) y.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape (n_samples, n_features)
+            The input data.
+
+        y : array-like, shape (n_samples,) or (n_samples, n_outputs)
+            The target values (class labels in classification, real numbers in
+            regression).
+
+        Returns
+        -------
+        self : returns a trained MLP model.
+        """
+        if getattr(self, 'classes_', None) is not None:
+            num_classes = len(unique_labels(y))
+            if num_classes != len(self.classes_):
```

Comment:
class size as the unique number of labels. For example unique(y) = [1,2,3] and unique(y_alt) = [4,5,6]. fit(X,y) then fit(X,y_alt) will not return an error.